python/cudf/cudf/core/frame.py

```diff
@@ -238,6 +239,45 @@ def _quantiles(
         result._copy_categories(self)
         return result
 
+    def rank(self, axis=0, method="average", numeric_only=None, na_option="keep", ascending=True, pct=False):
+        """
+        Compute numerical data ranks (1 through n) along axis.
+        By default, equal values are assigned a rank that is the average of the
+        ranks of those values.
+        Parameters
+        ----------
+        axis : {0 or 'index', 1 or 'columns'}, default 0
+            Index to direct ranking.
+        method : {'average', 'min', 'max', 'first', 'dense'}, default 'average'
+            How to rank the group of records that have the same value (i.e. ties):
+            * average: average rank of the group
+            * min: lowest rank in the group
+            * max: highest rank in the group
+            * first: ranks assigned in order they appear in the array
+            * dense: like 'min', but rank always increases by 1 between groups.
+        numeric_only : bool, optional
+            For DataFrame objects, rank only numeric columns if set to True.
+        na_option : {'keep', 'top', 'bottom'}, default 'keep'
+            How to rank NaN values:
+            * keep: assign NaN rank to NaN values
+            * top: assign smallest rank to NaN values if ascending
+            * bottom: assign highest rank to NaN values if ascending.
+        ascending : bool, default True
+            Whether or not the elements should be ranked in ascending order.
+        pct : bool, default False
+            Whether or not to display the returned rankings in percentile
+            form.
+        Returns
+        -------
+        same type as caller
+            Return a Series or DataFrame with data ranks as values.
+        """"
+        result = self._from_table(rank(
```

Comment:
This looks like a recursive call to `rank`. Let's just define `rank` at the frame level and get rid of `_sort.py:rank`.

Reply:
Thanks. updated as per comments.

Reply:
I need help with `numeric_only` argument. I saw `self.select(include=[np.number, np.bool])` could do it, but IIRC it creates a copy. of columns Since this is input read only, I don't want a copy created. What can be used to avoid copy for selecting numeric only columns?